src/game/client/components/items.cpp

```diff
@@ -51,8 +59,50 @@ void CItems::RenderProjectile(const CNetObj_Projectile *pCurrent, int ItemID)
 
 	vec2 StartPos(pCurrent->m_X, pCurrent->m_Y);
 	vec2 StartVel(pCurrent->m_VelX/100.0f, pCurrent->m_VelY/100.0f);
-	vec2 Pos = CalcPos(StartPos, StartVel, Curvature, Speed, Ct);
-	vec2 PrevPos = CalcPos(StartPos, StartVel, Curvature, Speed, Ct-0.001f);
+
+	vec2 Pos;
+	vec2 PrevPos;
+	if (pCurrent->m_Water)
+	{
+		if (pCurrent->m_Water == 1)
+		{
+			Pos = CalcPos(StartPos, StartVel, Curvature, Speed * WaterResistance, Ct);
+			PrevPos = CalcPos(StartPos, StartVel, Curvature, Speed * WaterResistance, Ct - 0.001f);
+
+			//vec2 PreWaterPos = CalcPos(vec2(pPrev->m_X, pPrev->m_Y), vec2(pPrev->m_VelX/100.0f, pPrev->m_VelY/100.0f), Curvature, Speed, Ct - 0.001f);
+			vec2 PreWaterPos = CalcPos(StartPos, StartVel, Curvature, Speed * WaterResistance, Ct - 0.001f);
+
+			// Need to figure out a smart way for this. Perhaps let server help out?
+			if (Collision()->IntersectWater(PreWaterPos, Pos, &PreWaterPos, vec2(1, 1)))
+			{
+				//m_pClient->m_pEffects->PowerupShine(PreWaterPos,vec2(1.0f,1.0f));
+				//Console()->Print(IConsole::OUTPUT_LEVEL_STANDARD, "game", "Water Surface Found");
+				if (m_pClient->m_pWater->HitWater(PreWaterPos.x, PreWaterPos.y, distance(PrevPos, Pos) * BulletSize * 25))
+				{
+					/*char aBuf[128];
+					char aBuf2[128];
+					char aBuf3[128];
+					str_format(aBuf, sizeof(aBuf), "%f", distance(PrevPos, Pos));
+					str_format(aBuf2, sizeof(aBuf2), "PreWater: %f  %f", PrevPos.x, PrevPos.y);
+					str_format(aBuf3, sizeof(aBuf3), "Pos: %f  %f", Pos.x, Pos.y);
+					Console()->Print(IConsole::OUTPUT_LEVEL_STANDARD, "game", aBuf);
+					Console()->Print(IConsole::OUTPUT_LEVEL_STANDARD, "game", aBuf2);
+					Console()->Print(IConsole::OUTPUT_LEVEL_STANDARD, "game", aBuf3);*/
```

Comment:
Better idea: If this is a debug print work with the DEBUG Macro

Reply:
I'll change it to Debug, since it does not need to be printed in the regular console :)
But I'll keep the code in for now, and remove it at the very end. It might come in useful. I will keep this unresolved so as not to forget!